### PR TITLE
Add source links to Kotlin DSL Reference docs

### PR DESCRIPTION
**Pull request by jbartok**
Fixes
* https://github.com/gradle/gradle/issues/24660

### CODE REVIEW

**Comment by jbartok** [comment]
@bot-gradle test this

**Comment by jbartok** [comment]
@bot-gradle test this

File: build-logic/documentation/src/main/groovy/gradlebuild/docs/GradleKotlinDslReferencePlugin.java [hunk 1]
```diff
@@ -88,15 +102,32 @@ private static void wireInArtificialSourceSet(Project project, GradleDocumentati
...
-        getDokkatooExtension(project).getPluginsConfiguration().named("html", DokkaHtmlPluginParameters.class, config -> {
-            config.getCustomStyleSheets().from(extension.getSourceRoot().file("kotlin/styles/gradle.css"));
-            config.getCustomAssets().from(extension.getSourceRoot().file("kotlin/images/gradle-logo.svg"));
-            config.getFooterMessage().set("Gradle Kotlin DSL Reference");
-        });
+    private static void configureSourceLinks(Project project, GradleDocumentationExtension extension, DokkaSourceSetSpec spec) {
+        extension.getSourceRoots().getFiles()
+            .forEach(
+                file -> {
+                    String path = file.getAbsolutePath();
```
Comment (eskatos) [inline build-logic/documentation/src/main/groovy/gradlebuild/docs/GradleKotlinDslReferencePlugin.java:113]: ❌  The path calculation should be made more correct and robust by relativizing against `project.rootDir`

**Review by eskatos — CHANGES_REQUESTED** [review]

File: build-logic/documentation/src/main/groovy/gradlebuild/docs/GradleKotlinDslReferencePlugin.java (hunk 1 above)
Comment (jbartok) [reply build-logic/documentation/src/main/groovy/gradlebuild/docs/GradleKotlinDslReferencePlugin.java]: Made the change, please take a look.

**Comment by eskatos** [comment]
@bot-gradle test BD

**Review by eskatos — APPROVED** [review]
🎉 

This should be added to the release notes!

**Review by lkasso — APPROVED** [review]

**Comment by jbartok** [comment]
@bot-gradle test and merge

**Comment by jbartok** [comment]
@bot-gradle test APT

**Comment by jbartok** [comment]
@bot-gradle test and merge

**Comment by jbartok** [comment]
@bot-gradle test this APT

**Comment by jbartok** [comment]
@bot-gradle test and merge

**Pull request merged**